localization/README.md

```diff
@@ -0,0 +1,183 @@
+# Localization
```

Comment:
simplify this documentation. 

- do not repeat aspnetcore documentation
- do not repeat what sample code already has
- only include our design decisions, docs links and sample links (or file names)